v3/MmlTree/ts/MmlNodes/mrow.ts

```diff
@@ -0,0 +1,213 @@
+    public get kind() {
+        return 'mrow';
+    }
+
+    /*
+     * An mrow is space-like if all its children are.
+     *
+     * @override
+     */
+    public get isSpacelike() {
+        for (const child of this.childNodes) {
+            if (!child.isSpacelike) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /*
+     * An mrow is embellished if it contains one embellished operator
+     * and any number of space-like nodes
+     *
+     * @override
+     */
+    public get isEmbellished() {
+        let embellished = false;
+        let i = 0;
+        for (const child of this.childNodes) {
+            if (child) {
+                if (child.isEmbellished) {
+                    if (embellished) {
+                        return false;
+                    }
+                    embellished = true;
+                    this._core = i;
+                } else if (!child.isSpacelike) {
+                    return false;
+                }
+            }
+        }
+        return embellished;
+    }
+
+    /*
+     * @override
+     */
+    public core(): MmlNode {
+        if (!this.isEmbellished || this._core == null) {
+            return this;
+        }
+        return this.childNodes[this._core];
+    }
+
+    /*
+     * @override
+     */
+    public coreMO(): MmlNode {
+        if (!this.isEmbellished || this._core == null) {
+            return this;
+        }
+        return this.childNodes[this._core].coreMO();
+    }
+
+    /*
+     * @return {number}  The number of non-spacelike child nodes
+     */
+    public nonSpaceLength() {
+        let n = 0;
+        for (const child of this.childNodes) {
+            if (child && !child.isSpacelike) {
+                n++;
+            }
+        }
+        return n;
+    }
+
+    /*
+     * @return {MmlNode}  The first non-space-like child node
+     */
+    public firstNonSpace() {
+        for (const child of this.childNodes) {
+            if (child && !child.isSpacelike) {
+                return child;
+            }
+        }
+        return null;
+    }
+
+    /*
+     * @return {MmlNode}  The last non-space-like child node
+     */
+    public lastNonSpace() {
+        let i = this.childNodes.length;
+        while (--i >= 0) {
+            let child = this.childNodes[i];
+            if (child && !child.isSpacelike) {
+                return child;
+            }
+        }
+        return null;
+    }
+
+    /*
+     * @override
+     */
+    public setTeXclass(prev: MmlNode) {
+        if ((this.getProperty('open') != null || this.getProperty('close') != null) &&
+            (!prev || prev.getProperty('fnOp') != null)) {
+            //
+            // <mrow> came from \left...\right
+            // so treat as subexpression (TeX class INNER).
+            //
+            this.getPrevClass(prev);
+            prev = null;
+            for (const child of this.childNodes) {
+                prev = child.setTeXclass(prev);
```

Comment:
> Why do we set TeXclass of the first element to null in this case?

We don't.  We set the *previous* element to null, because within `\left...\right`, the first child has no previous element to affect its spacing.

In TeX, the spacing is based on two adjacent items, and `setTeXclass(prev)` isn't setting the class to be `prev`, it is setting the TeX class given that the previous item is `prev`.  There is none in this case.

If that's not what you are meaning, let me know.